actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -234,7 +234,7 @@ def fragment_name_with_digest(name, virtual_path, digest_path)
         if virtual_path || digest_path
           name = controller.url_for(name).split("://").last if name.is_a?(Hash)
 
-          digest_path ||= digest_path_from_virtual(virtual_path)
+          digest_path ||= digest_path_from_template(@current_template)
```

Comment:
I'm taking this is the main difference? We've now captured the currently rendering template and can derive the format directly from it?

Reply:
Yep, exactly